### ISSUE TITLE
False positive E0213 no-self-argument for decorator which extends staticmethod

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#41]
Originally reported by: **BitBucket: [mitar](http://bitbucket.org/mitar), GitHub: @mitar?**

---

I have the following code:

```
#!python
def custom_decorator(platform=None, order=50):
    def wrapper(f):
        f.cgm_module = True
        f.cgm_module_order = order
        f.cgm_module_platform = platform
        return staticmethod(f)

    return wrapper

class SomeClass(object):
    @custom_decorator()
    def network(node, cfg):
        pass
```

And pylint complains that "Method should have "self" as first argument". But as this is a static method decorator in fact it should not need a "self" as first argument.

pylint 0.28.0, 
astng 0.24.3, common 0.59.1

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/41
